### PR TITLE
sched/init: move task list table from data to bss

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/init: move task list table from data to bss
sched/init: move idle task initialize to inline function

N/A, help to extend the multi-core implementation

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: sched/init/nx_start.c [hunk 1]
```diff
@@ -341,9 +323,9 @@ void nx_start(void)
        * that has pid == 0 and sched_priority == 0.
        */
 
-      memset((void *)&g_idletcb[i], 0, sizeof(struct task_tcb_s));
-      g_idletcb[i].cmn.pid        = i;
-      g_idletcb[i].cmn.task_state = TSTATE_TASK_RUNNING;
+      memset(tcb, 0, sizeof(struct task_tcb_s));
```
Comment (xiaoxiang781216) [inline sched/init/nx_start.c:366]: don't need memset again

File: sched/init/nx_start.c [hunk 2]
```diff
@@ -414,16 +395,117 @@ void nx_start(void)
...
+
+void nx_start(void)
+{
+  int i;
+
+  sinfo("Entry\n");
+
+  /* Boot up is complete */
+
+  nxsched_initstate() = OSINIT_BOOT;
```
Comment (xiaoxiang781216) [inline sched/init/nx_start.c:540]: call nxsched_set_initstate

Comment (anchao) [reply sched/init/nx_start.c]: Done

File: sched/init/nx_start.c (hunk 1 above)
Comment (anchao) [reply sched/init/nx_start.c:368]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by acassis — APPROVED** [review]

**Pull request merged**